(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py makemigrations
Migrations for 'dojos_ninjas_app':
  dojos_ninjas_app/migrations/0001_initial.py
    - Create model Dojo
    - Create model Ninja
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, dojos_ninjas_app, sessions
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying auth.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying admin.0003_logentry_add_action_flag_choices... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_alter_group_name_max_length... OK
  Applying auth.0011_update_proxy_permissions... OK
  Applying dojos_ninjas_app.0001_initial... OK
  Applying sessions.0001_initial... OK
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py shell
Python 3.9.2 (default, Mar 26 2021, 23:22:38) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.23.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from dojos_ninjas_app.models import *

In [2]: Dojo.objects.create(name="Yo",city="bellevue",state="WA")
Out[2]: <Dojo: Dojo object (1)>

In [3]: Dojo.objects.create(name="APP",city="Los Angeles",state="CA")
Out[3]: <Dojo: Dojo object (2)>

In [4]: Dojo.objects.create(name="coding",city="Pheonix",state="AZ")
Out[4]: <Dojo: Dojo object (3)>

In [5]: Dojo.objects.all().delete()
Out[5]: (3, {'dojos_ninjas_app.Ninja': 0, 'dojos_ninjas_app.Dojo': 3})

In [6]: Dojo.objects.all()
Out[6]: <QuerySet []>

In [7]: Dojo.objects.create(name="coding",city="Pheonix",state="AZ")
Out[7]: <Dojo: Dojo object (4)>

In [8]: Dojo.objects.create(name="APP",city="Los Angeles",state="CA")
Out[8]: <Dojo: Dojo object (5)>

In [9]: Dojo.objects.create(name="Yo",city="bellevue",state="WA")
Out[9]: <Dojo: Dojo object (6)>

In [10]: Ninja.objects.create(first_name="Tim",last_name="Koz",dojo=Dojo.objects.get(id=4))
Out[10]: <Ninja: Ninja object (1)>

In [11]: Ninja.objects.create(first_name="John",last_name="Smith",dojo=Dojo.objects.get(id=4))
Out[11]: <Ninja: Ninja object (2)>

In [12]: Ninja.objects.create(first_name="Ben",last_name="Poppy",dojo=Dojo.objects.get(id=4))
Out[12]: <Ninja: Ninja object (3)>

In [13]:     Ninja.objects.create(first_name="Jester",last_name="Matt",dojo=Dojo.objects.get(id=5))
Out[13]: <Ninja: Ninja object (4)>

In [14]: Ninja.objects.create(first_name="Klein",last_name="Honda",dojo=Dojo.objects.get(id=5))
Out[14]: <Ninja: Ninja object (5)>

In [15]: Ninja.objects.create(first_name="Artyom",last_name="Jack",dojo=Dojo.objects.get(id=5))
Out[15]: <Ninja: Ninja object (6)>

In [16]: Ninja.objects.create(first_name="Jacky",last_name="Liz",dojo=Dojo.objects.get(id=6))
Out[16]: <Ninja: Ninja object (7)>

In [17]: Ninja.objects.create(first_name="Fred",last_name="Flin",dojo=Dojo.objects.get(id=6))
Out[17]: <Ninja: Ninja object (8)>

In [18]: Ninja.objects.create(first_name="Jessica",last_name="Kat",dojo=Dojo.objects.get(id=6))
Out[18]: <Ninja: Ninja object (9)>

In [19]: Dojo.objects.ninjas.all()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-19-546105f5a664> in <module>
----> 1 Dojo.objects.ninjas.all()

AttributeError: 'Manager' object has no attribute 'ninjas'

In [20]: Dojo.objects.first().ninjas.all()
Out[20]: <QuerySet [<Ninja: Ninja object (1)>, <Ninja: Ninja object (2)>, <Ninja: Ninja object (3)>]>

In [21]: Dojo.objects.last().ninjas.all()
Out[21]: <QuerySet [<Ninja: Ninja object (7)>, <Ninja: Ninja object (8)>, <Ninja: Ninja object (9)>]>

In [22]: Ninja.object.last().dojo
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-22-6312c17057eb> in <module>
----> 1 Ninja.object.last().dojo

AttributeError: type object 'Ninja' has no attribute 'object'

In [23]: Ninja.objects.last().dojo
Out[23]: <Dojo: Dojo object (6)>

In [24]: exit
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py makemigrations
Traceback (most recent call last):
  File "/Users/tim/Documents/GitHub/Python_Stack/django/django_intro/dojos_/manage.py", line 21, in <module>
    main()
  File "/Users/tim/Documents/GitHub/Python_Stack/django/django_intro/dojos_/manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/Users/tim/Documents/GitHub/Python_Stack/my_environments/git_Django/lib/python3.9/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/Users/tim/Documents/GitHub/Python_Stack/my_environments/git_Django/lib/python3.9/site-packages/django/core/management/__init__.py", line 357, in execute
    django.setup()
  File "/Users/tim/Documents/GitHub/Python_Stack/my_environments/git_Django/lib/python3.9/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/tim/Documents/GitHub/Python_Stack/my_environments/git_Django/lib/python3.9/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/Users/tim/Documents/GitHub/Python_Stack/my_environments/git_Django/lib/python3.9/site-packages/django/apps/config.py", line 211, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/local/Cellar/python@3.9/3.9.2_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/tim/Documents/GitHub/Python_Stack/django/django_intro/dojos_/dojos_ninjas_app/models.py", line 11, in <module>
    class Ninja(models.Model):
  File "/Users/tim/Documents/GitHub/Python_Stack/django/django_intro/dojos_/dojos_ninjas_app/models.py", line 18, in Ninja
    Ninja.objects.create(first_name="Tim",last_name="Koz",dojo=Dojo.objects.get(id=4))
NameError: name 'Ninja' is not defined
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py makemigrations
No changes detected
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ ls
db.sqlite3              dojos_                  dojos_ninjas_app        manage.py
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py makemigrations
No changes detected
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, dojos_ninjas_app, sessions
Running migrations:
  No migrations to apply.
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py makemigrations
You are trying to add a non-nullable field 'desc' to dojo without a default; we can't do that (the database needs something to populate existing rows).
Please select a fix:
 1) Provide a one-off default now (will be set on all existing rows with a null value for this column)
 2) Quit, and let me add a default in models.py
Select an option: 1
Please enter the default value now, as valid Python
The datetime and django.utils.timezone modules are available, so you can do e.g. timezone.now
Type 'exit' to exit this prompt
>>> old dojo
Invalid input: unexpected EOF while parsing (<string>, line 1)
>>> 'old dojo'
Migrations for 'dojos_ninjas_app':
  dojos_ninjas_app/migrations/0002_dojo_desc.py
    - Add field desc to dojo
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, dojos_ninjas_app, sessions
Running migrations:
  Applying dojos_ninjas_app.0002_dojo_desc... OK
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ Dojo.objects.create(name="yellow",city="Bothell",state="WA",desc="the yellow one")
bash: syntax error near unexpected token `name="yellow",city="Bothell",state="WA",desc="the yellow one"'
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ Dojo.objects.last()
> quite
bash: syntax error near unexpected token `quite'
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py shell
Python 3.9.2 (default, Mar 26 2021, 23:22:38) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.23.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: Dojo.objects.create(name="yellow",city="Bothell",state="WA",desc="the yellow one")
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-1-29573167c5f0> in <module>
----> 1 Dojo.objects.create(name="yellow",city="Bothell",state="WA",desc="the yellow one")

NameError: name 'Dojo' is not defined

In [2]: Dojo.objects.create(name="APP",city="Los Angeles",state="CA")
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-2-ff7d171e801e> in <module>
----> 1 Dojo.objects.create(name="APP",city="Los Angeles",state="CA")

NameError: name 'Dojo' is not defined

In [3]: exit
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ ls
db.sqlite3              dojos_                  dojos_ninjas_app        manage.py
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py shell
Python 3.9.2 (default, Mar 26 2021, 23:22:38) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.23.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: Ninja.objects.last().dojo
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-1-23c5b6f8b7cf> in <module>
----> 1 Ninja.objects.last().dojo

NameError: name 'Ninja' is not defined

In [2]: exit
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py makemigrations
No changes detected
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py migrate
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, dojos_ninjas_app, sessions
Running migrations:
  No migrations to apply.
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ ls
db.sqlite3              dojos_                  dojos_ninjas_app        manage.py
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ cd dojos_ninjas_app/
(git_Django) Tanyas-MacBook-Pro:dojos_ninjas_app tim$ ls
__init__.py     __pycache__     admin.py        apps.py         migrations      models.py       save_shell.txt  tests.py        urls.py         views.py
(git_Django) Tanyas-MacBook-Pro:dojos_ninjas_app tim$ python manage.py shell
/Users/tim/Documents/GitHub/Python_Stack/my_environments/git_Django/bin/python: can't open file '/Users/tim/Documents/GitHub/Python_Stack/django/django_intro/dojos_/dojos_ninjas_app/manage.py': [Errno 2] No such file or directory
(git_Django) Tanyas-MacBook-Pro:dojos_ninjas_app tim$ cd ..
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py shell
Python 3.9.2 (default, Mar 26 2021, 23:22:38) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.23.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: Ninja.objects.last().dojo
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-1-23c5b6f8b7cf> in <module>
----> 1 Ninja.objects.last().dojo

NameError: name 'Ninja' is not defined

In [2]: exit
(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
May 28, 2021 - 02:21:01
Django version 2.2, using settings 'dojos_.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
^C(git_Django) Tanyas-MacBook-Pro:dojos_ tim$ python manage.py shell
Python 3.9.2 (default, Mar 26 2021, 23:22:38) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.23.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: Ninja.objects.last().dojo
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-1-23c5b6f8b7cf> in <module>
----> 1 Ninja.objects.last().dojo